src/Facade/Services/PcasStorageTypeFacadeService.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace Linn.Stores2.Facade.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Persistence;
+    using Linn.Stores2.Domain.LinnApps.Pcas;
+    using Linn.Stores2.Facade.Common;
+    using Linn.Stores2.Resources.Pcas;
+
+    public class PcasStorageTypeFacadeService : AsyncFacadeService<PcasStorageType, PcasStorageTypeKey, PcasStorageTypeResource, PcasStorageTypeResource, PcasStorageTypeResource>
+    {
+        public PcasStorageTypeFacadeService(
+            IRepository<PcasStorageType, PcasStorageTypeKey> repository,
+            ITransactionManager transactionManager,
+            IBuilder<PcasStorageType> resourceBuilder)
+            : base(repository, transactionManager, resourceBuilder)
+        {
+        }
+
+        protected override async Task<PcasStorageType> CreateFromResourceAsync(
+            PcasStorageTypeResource resource,
+            IEnumerable<string> privileges = null)
+        {
+            return new PcasStorageType(
+                resource.BoardCode,
+                resource.StorageTypeCode,
```

Comment:
actually on second thought, I would keep this async. reason being I think for this to work at all you are going to need to look up the board and the storage type from the db with async operations. I would change thise code to not allow passing any string for BoardCode and StorageTypeCode - instead you should be passing an existing Board and StorageType that result from a repository lookup